maven/spec/dependabot/maven/version_spec.rb

```diff
@@ -115,295 +145,350 @@
 
       it { is_expected.to be(false) }
     end
+  end
 
-    context "with a 'pr' pre-release separated with a ." do
```

Comment:
@abdulapopoola @jonjanego I could do with your help here. The maven spec: https://maven.apache.org/pom.html#version-order-specification does not say that versions with 'pr' or 'dev' in them are prereleases but in our current maven version implementation we are categorising them as such. Should I stick to the spec, in which case I'll remove these tests as I already have done, or should I add this to the new implementation as an 'addition' to the spec to avoid breaking changes for our clients? 

In my view, it's better to strictly follow the spec since standardising is one of the goals but I think this is a product decision so I'd like to hear your views.

Reply:
Hi @amazimbe -- I agree with you, especially since those scenarios implied by those qualifiers have "official" alternatives:

> "alpha" < "beta" < "milestone" < "rc" = "cr" < "snapshot" < "" = "final" = "ga" < "sp"

do we have any metrics on usage of those qualifiers?

Reply:
I doubt we have any metrics for these but I can ask around.

Reply:
For what it's worth, `pre` was only added [very recently](https://github.com/dependabot/dependabot-core/pull/10207), so it's probably not used a lot, although the user who requested supporting it probably won't be happy.

I also think it's better to follow the spec strictly (and I rejected a few requests to add more qualifiers in the past), but note that not even maven itself always follows its own spec. See https://issues.apache.org/jira/browse/MNG-7690.

Reply:
yes, I think that we should give them a heads up.

Reply:
sorry my bad, I approved that PR without knowing what I currently know. I still think it's better to follow the spec 100% and we should reset expectations going forward.

Good catch @deivid-rodriguez !

Reply:
No problem, just wanted to make sure you were all on the same page :)